core/services/relay/evm/evm.go

```diff
@@ -255,6 +260,21 @@ func (r *Relayer) NewMercuryProvider(rargs commontypes.RelayArgs, pargs commonty
 		clients[server.URL] = client
 	}
 
+	// initialize trigger capability service lazily
```

Comment:
does this need to be guarded with some sort of singleton logic or is Add idempotent? NewMercuryProvider can be  called in multiple times

Reply:
Yeah this is guarded by "r.triggerCapability == nil" check and I believe NewMercuryProvider() will not be called in parallel (because delegate.ServicesForSpec() isn't)